Sources/Error Handling/PurchasesError.swift

```diff
@@ -65,3 +88,42 @@ extension PurchasesError {
     }
 
 }
+
+@available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, visionOS 1.0, *)
+private extension PurchasesError {
+
+    func getStoreKitErrorInfoIfAny(error: Error) -> [String: Any]? {
+        if let skError = error as? SKError {
+            return [
+                "skErrorCode": skError.code,
+                "description": skError.code.trackingDescription
+            ]
+        } else if let storeKitError = error as? StoreKitError {
+            let resultMap: [String: Any] = ["description": storeKitError.trackingDescription]
+            switch storeKitError {
+            case .unknown,
+                    .userCancelled,
+                    .notAvailableInStorefront,
+                    .notEntitled:
+                return resultMap
+            case let .networkError(urlError):
+                return resultMap.merging([
+                    "urlErrorCode": urlError.errorCode,
+                    "urlErrorFailingUrl": urlError.failureURLString ?? ""
+                ])
+            case let .systemError(systemError):
+                return resultMap.merging([
+                    "systemErrorDescription": systemError.localizedDescription
+                ])
+
+            @unknown default:
+                return resultMap
+            }
+        } else if let storeKitError = error as? StoreKit.Product.PurchaseError {
+            return ["description": storeKitError.trackingDescription]
+        } else {
+            return nil
```

Comment:
I guess it would be useful to log what other error type we might be missing? 🤔

Reply:
Right, in this case I'm assuming it's because the root error is not a store kit error (which may happen in several situations). I could add a log here, but I think the situation may happen relatively often... Note that this method is only to add extra data to the `userInfo` map in case it's actually a StoreKit error.